recipes/gazebo-yarp-plugins/meta.yaml

```diff
@@ -0,0 +1,54 @@
+{% set name = "libgazebo-yarp-plugins" %}
+{% set version = "4.6.0" %}
+
+package:
+  name: {{ name }}
+  version: {{ version }}
+
+source:
+  - url: https://github.com/robotology/gazebo-yarp-plugins/archive/refs/tags/v{{ version }}.tar.gz
+    sha256: 26e4f26ca5799f445bef9ad5ef307da4b180bebe593a1c925246598eb61b8521
+
+build:
+  number: 0
+  skip: true  # [osx]
+  run_exports:
+    - {{ pin_subpackage(name, max_pin='x') }}
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - cmake
+    - pkg-config
+    - ninja
+    - multisheller
+  host:
+    - yarp-cxx
+    - gazebo
+    - libsdformat
+    - libignition-math6
+    - libprotobuf
+    - tbb-devel
+    - libopencv
+    - ogre
+  run:
+    - __osx >={{ MACOSX_DEPLOYMENT_TARGET|default("10.9") }}  # [osx and x86_64]
+
+test:
+  commands:
+    - test -f ${PREFIX}/include/GazeboYarpPlugins/common.h  # [not win]
+    - test -f ${PREFIX}/lib/libgazebo_yarp_camera${SHLIB_EXT}  # [unix]
+    - if not exist %LIBRARY_INC%\\GazeboYarpPlugins\\common.h exit 1  # [win]
+    - if not exist %LIBRARY_BIN%\\gazebo_yarp_camera.dll exit 1  # [win]
+
```

Comment:
I strongly recommend checking existence for all your shared libraries and some CMake files. You can use a loop in jinja to reduce the amount of boiler plate for the shared objects since they all start with the same `$PREFIX/lib/libgazebo_yarp_`.

something like 

```yaml
{%for module in ['camera', 'foo'] %}
    - test -f ${PREFIX}/lib/libgazebo_yarp_{{ module }}${SHLIB_EXT}    # [unix]
    - if not exist %LIBRARY_BIN%\\gazebo_yarp_{{ module }}.dll exit 1  # [win]
{%endfor%}
```

There's an example in the [tk feedstock](https://github.com/conda-forge/tk-feedstock/blob/ac2bc60277cc4769d66a63be00a7650feca6df30/recipe/meta.yaml).

Reply:
Cool, I was not aware of it.

Reply:
I tried to address your comment in https://github.com/conda-forge/staged-recipes/pull/21891/commits/192a30df60153efae56fd525a86e79baf3ea3e1c . @carterbox